src/hotspot/share/interpreter/interpreterRuntime.cpp

```diff
@@ -225,17 +225,11 @@ JRT_ENTRY(void, InterpreterRuntime::_new(JavaThread* current, ConstantPool* pool
 
   // At this point the class may not be fully initialized
   // because of recursive initialization. If it is fully
-  // initialized & has_finalized is not set, we rewrite
-  // it into its fast version (Note: no locking is needed
-  // here since this is an atomic byte write and can be
-  // done more than once).
+  // initialized we rewrite it into its fast version
+  // (Note: no locking is needed here since this is an
+  // atomic byte write and can be done more than once).
   //
-  // Note: In case of classes with has_finalized we don't
-  //       rewrite since that saves us an extra check in
-  //       the fast version which then would call the
-  //       slow version anyway (and do a call back into
-  //       Java).
-  //       If we have a breakpoint, then we don't rewrite
+  // Note: If we have a breakpoint, then we don't rewrite
```

Comment:
I don't know what this comment is referring to.  We don't rewrite the 'new' bytecode into a faster version, which is what I think it is saying. I'd rather see this comment deleted than try to figure out what it might be saying...

Reply:
Perfect.  I was hoping that would be the outcome as I couldn't find anything else that made sense for the rest of the comment either

Reply:
Even the bit where the "At this point the class might not be fully initialized".  It might be creating an instance during initialization, I guess.  But I don't see how this comment helps anyone understand anything in this code.